students/eowyn/backcast.py

```diff
@@ -0,0 +1,252 @@
+import pandas as pd
+import numpy as np
+from tabulate import tabulate
+
+fname = 'M426_Processed_Data.txt'
+metdf = pd.read_table(fname, skiprows=1,
+                      index_col=0,
+                      parse_dates=True,
+                      infer_datetime_format=True)
+metdf.index.name = 'DateTime'
+
+"""
+Parse column names to determine sensor info: measurement height,
+sensor type, and field type (mean, max, etc.) handling the existance
+of fields like "St Dev" that need to be combined to "StDev" and
+retaining only the numeric component of the sensor height (in m or ft)
+"""
+colnames = list(metdf)
+sensorInfo = []
+for col in colnames:
+    sensor = col.split(' ')
+    unitIndex, conversion = sensor[0].lower().find('m'), 1.0
+    if unitIndex == -1:
+        unitIndex, conversion = sensor[0].lower().find('ft'), 0.3048
```

Comment:
ouch! semi-ugly dataset!

probably not worth it for this but there are unit packages for python. none are perfect, but "pint" is pretty good.

https://pint.readthedocs.io/en/latest/